### ISSUE TITLE
Resource Limits are more than just a limit

### ISSUE THREAD

**Issue by tpretz** [truenas/charts#948]
Hi,
Have a bunch of containers running.

Wanted to add resource limits, just to ensure a badly behaved container cannot burst and consume all resources.
More of safeguards, set fairly high.

After setting a few such limits, successfullly i found apps began getting stuck in "deployment" stage.

Had a debug via kubectl, essentially pods were getting stuck waiting for placement (not enough resource to deploy), which was unexpected.

Checking out the helm chart docs, https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/, they do specify:

> Note: If you specify a limit for a resource, but do not specify any request, and no admission-time mechanism has applied a default request for that resource, then Kubernetes copies the limit you specified and uses it as the requested value for the resource.

This seems to be where the issue is coming from, my fairly high limits are being applied as reservations and as the sum of all limits is an overcommit k8s refuses to deploy more pods.

Long term this may be fixed by https://github.com/truenas/charts/pull/917 as its refactoring the limits code and adding explicit request templating.

Shorter term im going to look to override locally and set a low / zero resource request along with the limits by adjusting https://github.com/truenas/charts/blob/master/library/common/2207.0.0/templates/lib/resources/_limitations.tpl

You may want to look at a fix for this prior to 23.10, especially if there is a way of templating in "use the default request reservation" when you add the limits entires

**Comment by Ornias1993** [comment]
If you want reliable resource limit settings, use TrueCharts.

**Comment by tpretz** [comment]
Yeah i have worked around it by adding a clone catalog item of ix-chart so i can migrate like 4 like for now.

I do use truecharts for a few things, but sometimes (mostly for older things that need a redeploy now charts exist) i would use the docker container launch button, so they are stuck on ix-chart till i invest a little more time.

Given it's an official option in the apps page docker wizard, regardless of better alternatives i still think its worth a fix.
Either by setting a low / removed resource request
Or, with a wizard text update to make it clear it is a resource reservation, not just a limit.

**Comment by rehevkor5** [comment]
It's also not clear how to use the `common` chart to give users control over Requests separately from Limits at the level of different Pods. For example, the value might be at `.Values.workload.elasticsearch.podSpec.containers.diskoverdata.resources.requests.cpu`. Does the `questions.yaml` need to contain 8 levels of nested dicts in order to make that work? Or is there some non-obvious easy way to do that more gracefully?

**Comment by stavros-k** [comment]
Yes and no.

But, I'll start saying that current elasticsearch / diskoverdata does **not** have that option yet,as it uses the old common still.

Back to the initial question. There are more than 1 right ways to do it.
1. re-create that dictionary in questions.yaml
2. have that dictionary in the templates and make it pull values from any dict from questions/values.yaml ( See for example the new Chia app)
